classes/blocks/class-oldenform.php

```diff
@@ -0,0 +1,378 @@
+				'thankyou_subtitle'             => '',
+				'thankyou_social_media_message' => '',
+				'thankyou_donate_message'       => '',
+				'donate_button_checkbox'        => '',
+				'thankyou_take_action_message'  => '',
+				'thankyou_url'                  => '',
+				'background'                    => '',
+				'en_form_id'                    => '',
+			],
+			$attributes,
+			'shortcake_enform'
+		);
+
+		return $this->render( $attributes );
+	}
+
+	/**
+	 * Get all the data that will be needed to render the block correctly.
+	 *
+	 * @param array $attributes This is the array of fields of this block.
+	 *
+	 * @return array The data to be passed in the View.
+	 */
+	public function prepare_data( $attributes ): array {
+
+		global $post;
+
+		// Extract twitter account from footer.
+		$social_menu = wp_get_nav_menu_items( 'Footer Social' );
+		if ( isset( $social_menu ) && is_iterable( $social_menu ) ) {
+			foreach ( $social_menu as $social_menu_item ) {
+				$url_parts = explode( '/', rtrim( $social_menu_item->url, '/' ) );
+				if ( false !== strpos( $social_menu_item->url, 'twitter' ) ) {
+					$social_accounts['twitter'] = count( $url_parts ) > 0 ? $url_parts[ count( $url_parts ) - 1 ] : '';
+				}
+			}
+		}
+
+		// Handle background image.
+		if ( isset( $attributes['background'] ) ) {
+			$options                         = get_option( 'planet4_options' );
+			$p4_happy_point_bg_image         = $options['happy_point_bg_image_id'] ?? '';
+			$image_id                        = '' !== $attributes['background'] ? $attributes['background'] : $p4_happy_point_bg_image;
+			$img_meta                        = wp_get_attachment_metadata( $image_id );
+			$attributes['background_src']    = wp_get_attachment_image_src( $image_id, 'retina-large' );
+			$attributes['background_srcset'] = wp_get_attachment_image_srcset( $image_id, 'retina-large', $img_meta );
+			$attributes['background_sizes']  = wp_calculate_image_sizes( 'retina-large', null, null, $image_id );
+		}
+		$attributes['default_image'] = get_bloginfo( 'template_directory' ) . '/images/happy-point-block-bg.jpg';
+
+		if ( is_object( $post ) ) {
+			$og_title       = get_post_meta( $post->ID, 'p4_og_title', true );
+			$og_description = get_post_meta( $post->ID, 'p4_og_description', true );
+			$link           = get_permalink( $post->ID );
+		} else {
+			$og_title       = '';
+			$og_description = '';
+			$link           = '';
+		}
+
+		if ( '' === $og_title && is_object( $post ) ) {
+			$title = get_the_title( $post->ID );
+			if ( '' !== $title ) {
+				$og_title = $title;
+			}
+		}
+		$social = [
+			'title'       => esc_attr( $og_title ),
+			'description' => esc_attr( wp_strip_all_tags( $og_description ) ),
+			'link'        => esc_url( $link ),
+		];
+
+		$data = [];
+
+		if ( isset( $attributes['thankyou_url'] ) && $attributes['thankyou_url'] && 0 !== strpos( $attributes['thankyou_url'], 'http' ) ) {
+			$attributes['thankyou_url'] = 'http://' . $attributes['thankyou_url'];
```

Comment:
SonarCloud [signals this as a security issue](https://sonarcloud.io/project/security_hotspots?id=greenpeace_planet4-plugin-gutenberg-blocks&pullRequest=597&resolved=false&types=SECURITY_HOTSPOT). Now this isn't really changed by this PR, so we could ignore it. But unless there's a specific good reason for using `http`, we probably should redirect to `https` here.

Reply:
Perhaps @comzeradd knows whether `http` is intentional here?

Reply:
Since SonarCloud allows to, I marked the security issue as "safe", and created a [dedicated ticket](https://jira.greenpeace.org/browse/PLANET-6193) to look into it.

Reply:
I don't think there was specific reason for that. We could just change that to `https`, since we only add that when people forget to add a protocol to the url.